### ISSUE TITLE
More description and samples for Task.Run overloads

### ISSUE THREAD

**Issue by KirillOsenkov** [dotnet/docs#3725]
For Task.Run overloads it would be nice if the semantics was explained in a bit more detail:
https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.run?view=netframework-4.7.1

Specifically for `Task Task.Run(Func<Task> t)` there is no actual description:
https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.run?view=netframework-4.7.1#System_Threading_Tasks_Task_Run_System_Func_System_Threading_Tasks_Task__

One valuable thing to have is a code sample that is roughly equivalent, such as:
```
    public static async Task Run(Func<Task> func) 
    {    
        Task t = null;
        await Task.Run(() => { t = func(); });
        await t;
    }
```
https://sharplab.io/#v2:D4AQDABCCMDcCwAocVoFYGKSAzFATBAMIRIDeSEVUeMAbFABxQMBKArgHYAUIaAPCDoA+CADMuAYwCUpRNQhkFlBdSEQALhAC8ETuwA2BzKrUBOFgDoOPbrO2ilW3RM6S7sCAF9pJ01AsNPyovJC8gA=

Basically it should explain what task unwrapping is, how it schedules one task to wait for the method to return, then waits on the task returned by that method and only then the outer task completes.

**Comment by mairaw** [comment]
Thanks for filing the issue @KirillOsenkov. @stephentoub would you be the right person to sync on this?

**Comment by stephentoub** [comment]
> would you be the right person to sync on this?

Sure
